pkg/oci/registry/client.go

```diff
@@ -116,10 +116,21 @@ func (c *Client) GetImageVersion(ctx context.Context, imageName string) (ImageVe
 	if err != nil {
 		return ImageVersion{}, fmt.Errorf("descriptor.Image(): %w", err)
 	}
-	dig, err := img.Digest()
+
+	// use image digest as a fallback
+	digestFn := img.Digest
+
+	// try to get image manifest to cover multi arch images
+	digestSrc, err := descriptor.ImageIndex()
```

Comment:
you are re-assigning the error variable here that means the error check in 130 does not make sense anymore or am I missing something

Reply:
The error check in 130 is for the outcome of 129, Or am I understanding something wrong?

Reply:
we do not care about the error from `descriptor.ImageIndex()`, as that errors (in theory) when not a image-index was given but an actual image (like `snapshot-amd64` instead of just `snapshot`)

the actual error that we care about, and should be propagated,  is what the `digestFn` would return, as that shouldn't error

TLDR: This is fine, BUT pls rename `digestSrc` to `imageIndex` as `digestSrc` is veeery generic 😅